src/panels/lovelace/editor/config-elements/hui-statistics-graph-card-editor.ts

```diff
@@ -297,18 +297,29 @@ export class HuiStatisticsGraphCardEditor
   }
 
   private async _entitiesChanged(ev: CustomEvent): Promise<void> {
-    const config = { ...this._config!, entities: ev.detail.value };
+    const newEntityIds = ev.detail.value;
+
+    // Save the EntityConfig objects from being replaced with strings
+    const newEntities = newEntityIds.map((newEnt) => {
+      const matchEntities = this._config!.entities.filter(
```

Comment:
```suggestion
      const matchEntities = this._config!.entities.find(
```

Reply:
Use find, so it stops when the item you are looking for was found

Reply:
done